app/presenters/tree_node_builder.rb

```diff
@@ -62,108 +69,126 @@ def build
-                                                _("Button: %{button_description}") % {:button_description => object.description})
-    when CustomButtonSet      then custom_button_set_node
-    when CustomizationTemplate then generic_node(object.name, "customizationtemplate.png")
-    when Dialog               then generic_node(object.label, "dialog.png")
-    when DialogTab            then generic_node(object.label, "dialog_tab.png")
-    when DialogGroup          then generic_node(object.label, "dialog_group.png")
-    when DialogField          then generic_node(object.label, "dialog_field.png")
-    when EmsFolder            then ems_folder_node
-    when EmsCluster           then generic_node(object.name, "cluster.png", "#{ui_lookup(:table => "ems_cluster")}: #{object.name}")
-    when GuestDevice          then guest_node(object)
-    when Host                 then generic_node(object.name,
+                                                _("Button: %{button_description}") % {:button_description => object.description}) },
+    "CustomButtonSet"            => -> { custom_button_set_node },
+    "CustomizationTemplate"      => -> { generic_node(object.name, "customizationtemplate.png") },
+    "Dialog"                     => -> { generic_node(object.label, "dialog.png") },
+    "DialogTab"                  => -> { generic_node(object.label, "dialog_tab.png") },
+    "DialogGroup"                => -> { generic_node(object.label, "dialog_group.png") },
+    "DialogField"                => -> { generic_node(object.label, "dialog_field.png") },
+    "EmsFolder"                  => -> { ems_folder_node },
+    "EmsCluster"                 => -> { generic_node(object.name, "cluster.png", "#{ui_lookup(:table => "ems_cluster")}: #{object.name}") },
+    "GuestDevice"                => -> { guest_node(object) },
+    "Host"                       => -> { generic_node(object.name,
                                                 "host.png",
-                                                "#{ui_lookup(:table => "host")}: #{object.name}")
-    when IsoDatastore         then generic_node(object.name, "isodatastore.png")
-    when IsoImage             then generic_node(object.name, "isoimage.png")
-    when ResourcePool         then generic_node(object.name, object.vapp ? "vapp.png" : "resource_pool.png")
+                                                "#{ui_lookup(:table => "host")}: #{object.name}") },
+    "IsoDatastore"               => -> { generic_node(object.name, "isodatastore.png") },
+    "IsoImage"                   => -> { generic_node(object.name, "isoimage.png") },
+    "ResourcePool"               => -> { generic_node(object.name, object.vapp ? "vapp.png" : "resource_pool.png") },
 
-    when Vm                   then vm_node(object)
-    when Lan                  then generic_node(object.name,
+    "Vm"                         => -> { vm_node(object) },
```

Comment:
I guess I don't follow with this one.  Is this something that is failing in the tests (haven't looked at them yet)?  didn't really change anything here and the only thing I could think is a problem is `obeject.class.to_s` never maps to `"Vm"`.  Is that what you meant?